src/anndata/_io/specs/methods.py

```diff
@@ -109,7 +112,9 @@ def wrapper(
 @_REGISTRY.register_read(H5File, IOSpec("", ""))
 @_REGISTRY.register_read(H5Group, IOSpec("", ""))
 @_REGISTRY.register_read(H5Array, IOSpec("", ""))
-def read_basic(elem, _reader):
+def read_basic(
+    elem: StorageType, _reader: Reader
+) -> dict | np.ndarray[Any, Any] | np.ndarray | sparse.spmatrix | SpArray:
```

Comment:
Not sure about the meaningfulness of `np.ndarray[Any, Any] | np.ndarray` and the like - need to look into it